src/libvirtApi/common.js

```diff
@@ -161,88 +165,103 @@ function doUsagePolling(name, connectionName, objPath) {
+        if (Object.getOwnPropertyNames(info).length > 0) {
+            const props = { name, connectionName, id: objPath };
+            let avgvCpuTime = 0;
+
+            if ("balloon.rss" in info)
+                props.rssMemory = info["balloon.rss"].v.v;
+            else if ("state.state" in info && info["state.state"].v.v == Enum.VIR_DOMAIN_SHUTOFF)
+                props.rssMemory = 0.0;
+            for (let i = 0; i < info["vcpu.maximum"].v.v; i++) {
+                if (!(`vcpu.${i}.time` in info))
+                    continue;
+                avgvCpuTime += info[`vcpu.${i}.time`].v.v;
+            }
+            avgvCpuTime /= info["vcpu.current"].v.v;
+            if (info["vcpu.current"].v.v > 0)
+                Object.assign(props, {
+                    actualTimeInMs: Date.now(),
+                    cpuTime: avgvCpuTime
+                });
+            Object.assign(props, {
+                disksStats: calculateDiskStats(info)
+            });
+
+            logDebug(`doUsagePolling: ${JSON.stringify(props)}`);
+            store.dispatch(updateVm(props));
+        }
+    } catch (ex) {
+        console.warn(`GetStats(${name}, ${connectionName}) failed: ${ex.toString()}`);
+    }
+    delayPolling(() => doUsagePolling(name, connectionName, objPath), null);
 }
 
-function getLoggedInUser() {
-    logDebug(`GET_LOGGED_IN_USER:`);
-    return cockpit.user().then(loggedUser => store.dispatch(setLoggedInUser({ loggedUser })));
+async function getLoggedInUser() {
+    const loggedUser = await cockpit.user();
+    logDebug(`GET_LOGGED_IN_USER:`, loggedUser);
+    store.dispatch(setLoggedInUser({ loggedUser }));
 }
 
-export function getLibvirtVersion({ connectionName }) {
-    return call(connectionName, "/org/libvirt/QEMU", "org.freedesktop.DBus.Properties", "Get", ["org.libvirt.Connect", "LibVersion"], { timeout, type: "ss" })
-            .then(version => store.dispatch(updateLibvirtVersion({ libvirtVersion: version[0].v })));
+export async function getLibvirtVersion({ connectionName }) {
+    const [version] = await call(connectionName, "/org/libvirt/QEMU", "org.freedesktop.DBus.Properties", "Get", ["org.libvirt.Connect", "LibVersion"], { timeout, type: "ss" });
+    store.dispatch(updateLibvirtVersion({ libvirtVersion: version.v }));
 }
 
-function getNodeMaxMemory({ connectionName }) {
-    // Some nodes don"t return all memory in just one cell.
+async function getCapabilities({ connectionName }) {
+    try {
+        const [capabilitiesXML] = await call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "GetCapabilities", [],
+                                             { timeout, type: "" });
+        const capabilities = parseDumpxmlForCapabilities(capabilitiesXML);
+        store.dispatch(setCapabilities({ capabilities }));
+    } catch (ex) {
+        console.warn("NodeGetMemoryStats failed:", ex.toString());
+        throw ex;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1280-20240125-124532-d8538f72-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)